### DISCUSSION TITLE
Server side error 500

### DISCUSSION THREAD

**Discussion by LongMingWei** [vercel/community#5070] [Help]
### Summary

When I build the project, the sign out button works fine, it logs you out of your account. However, in Vercel, it doesn't work. 
This was the message in the inspect console when pressing the sign out button in the page after logging in: 
![Screenshot 2023-12-05 124445](https://github.com/vercel/community/assets/131566573/7046c77b-7857-4c27-9644-63182631ff5c)

This was the error log in Vercel, 
![Screenshot 2023-12-05 124158](https://github.com/vercel/community/assets/131566573/e39ca7d6-a835-41ce-a411-9f2cb1d20a31)
![Screenshot 2023-12-05 124245](https://github.com/vercel/community/assets/131566573/45c32a42-8273-4aef-a6c2-bcd49e246ae1)

These were the code responsible for the signing in, signing out logic

    ` 'use server'
    import NextAuth from 'next-auth';
    import Credentials from 'next-auth/providers/credentials';
    import { authConfig } from './auth.config';
    import { z } from 'zod';
    import type { User } from './definitions';
    import bcrypt from 'bcrypt';
    import { connectToDatabase } from '@/components/mongo';
     
    async function getUser(name: string) {
      try {
        const db = await connectToDatabase();
        const users = db.collection('users');
        const user = await users.findOne({ name }, { projection: { name: 1, password: 1, _id: 0 } });
        if (user) return user;
      } catch (error) {
        console.error('Failed to fetch user:', error);
        throw new Error('Failed to fetch user.');
      }
      return undefined;
    }
     
    export const { auth, signIn, signOut } = NextAuth({
      ...authConfig,
      providers: [
        Credentials({
          async authorize(credentials) {
            const parsedCredentials = z
              .object({ name: z.string(), password: z.string() })
              .safeParse(credentials);
        
        if (!parsedCredentials.success) {
          console.error(parsedCredentials.error.errors);
        }
        if (parsedCredentials.success) {
          const { name, password } = parsedCredentials.data;
          const user = await getUser(name);
          if (!user) return null;
          const passwordsMatch = password.trim() === user.password.trim()
          if (passwordsMatch) return { _id: "", email: "", name: user.name };
        }
 
        console.log('Invalid');
        return null;
      },
    }),
    ],
    });
`

    `import type { NextAuthConfig } from 'next-auth';
    
    export const authConfig: NextAuthConfig = {
      pages: {
        signIn: '/',
      },
      callbacks: {
        authorized({ auth, request: { nextUrl } }) {
          const isLoggedIn = !!auth?.user;
          const isOnChat = nextUrl.pathname.startsWith('/chat');
  
        if (isOnChat) {
          if (isLoggedIn) {
            const currentUsername = auth?.user?.name;
            const requestedUsername = nextUrl.pathname.split('/').pop();
  
            if (currentUsername === requestedUsername) {
              return true;
            } else {
              return Response.redirect(new URL(`/chat/${currentUsername}`, nextUrl));
            }
          }
  
          return false;
        } else if (isLoggedIn) {
          return Response.redirect(new URL(`/chat/${auth?.user?.name}`, nextUrl));
        }
  
        return true;
      },
    },

    providers: [], 
      }; `

### Example

https://telegrammer-ebon.vercel.app/

### Steps to Reproduce

After clicking the link, sign in to a test account (Username: Test, Password: 123), then try to sign out. Right click, click on Inspect and the errors will show up in the console.

**Comment by amyegan** [comment]
Hey @LongMingWei. I have a few things to check that may help narrow down the cause of this error.

- Does the sign in button work when the project is deployed? (That might seem like a silly question, but I don't want to make any assumptions about the situation)
- Missing or mismatched env vars are a common cause of errors. Are the same [environment variables](https://vercel.com/docs/projects/environment-variables) set for development and preview? 

If you haven't already, I recommend also reaching out to NextAuth.js folks at [github.com/nextauthjs/next-auth/discussions](https://github.com/nextauthjs/next-auth/discussions). They know more about the specific configuration requirements of the package and may recognize what's going wrong.

**Reply by LongMingWei** [reply]
Thanks for replying! The sign in button does indeed work when the project is deployed, in fact every functionality works except the sign out button. I have also checked the env variables for both local and vercel, and they both match. Or do you mean something else when you say env variables for deployment and preview? 

Thank you!
